src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs

```diff
@@ -2520,36 +2532,55 @@ internal static void WriteConsole(ConsoleHandle consoleHandle, string output)
 
             while (cursor < output.Length)
             {
-                string outBuffer;
+                ReadOnlySpan<char> outBuffer;
 
                 if (cursor + maxBufferSize < output.Length)
                 {
-                    outBuffer = output.Substring(cursor, maxBufferSize);
+                    outBuffer = output.Slice(cursor, maxBufferSize);
                     cursor += maxBufferSize;
                 }
                 else
                 {
-                    outBuffer = output.Substring(cursor);
+                    outBuffer = output.Slice(cursor);
                     cursor = output.Length;
                 }
 
-                DWORD charsWritten;
-                bool result =
-                    NativeMethods.WriteConsole(
-                        consoleHandle.DangerousGetHandle(),
-                        outBuffer,
-                        (DWORD)outBuffer.Length,
-                        out charsWritten,
-                        IntPtr.Zero);
-
-                if (result == false)
+                if (newLine)
                 {
-                    int err = Marshal.GetLastWin32Error();
-
-                    HostException e = CreateHostException(err, "WriteConsole",
-                        ErrorCategory.WriteError, ConsoleControlStrings.WriteConsoleExceptionTemplate);
-                    throw e;
+                    Span<char> outBufferLine = stackalloc char[outBuffer.Length + 2];
+                    outBuffer.CopyTo(outBufferLine);
+                    ConsoleHostUserInterface.Crlf.AsSpan().CopyTo(outBufferLine.Slice(outBufferLine.Length-2));
```

Comment:
Same here for `2` and also different whitespace for subtraction than addition above

Reply:
Fixed.